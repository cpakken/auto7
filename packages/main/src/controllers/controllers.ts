import { ILogicController, ITypeController } from "."

export interface IControllers {
  logic: ILogicController
  type: ITypeController
}
