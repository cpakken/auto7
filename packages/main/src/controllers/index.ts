export * from "./types"
export * from "./logic"
export * from "./controllers"
