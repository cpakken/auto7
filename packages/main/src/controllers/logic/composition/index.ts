export * from "./connections-map"
export * from "./paths"
export * from "./blocks"
export * from "./composition"
