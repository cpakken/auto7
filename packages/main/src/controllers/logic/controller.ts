import { getEnv, getRoot, IAnyStateTreeNode, Instance, SnapshotIn, SnapshotOut, types } from "mobx-state-tree"
import { LogicComposedStore, LogicNativeStore } from "."

export interface ILogicControllerSnapshotIn extends SnapshotIn<typeof LogicController> {}
export interface ILogicControllerSnapshot extends SnapshotOut<typeof LogicController> {}
export interface ILogicController extends Instance<typeof LogicController> {}
export const LogicController = types.model({
  composed: types.optional(LogicComposedStore, {}),
  native: types.optional(LogicNativeStore, {}),
})

export function getLogicController(instance: IAnyStateTreeNode): ILogicController {
  return getEnv(instance).logicController || getRoot(instance)
}
