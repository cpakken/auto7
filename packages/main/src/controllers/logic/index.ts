export * from "./info"
export * from "./composition"
export * from "./stores"
export * from "./controller"
