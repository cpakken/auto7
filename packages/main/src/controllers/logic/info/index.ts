export * from "./interface"
export * from "./info"