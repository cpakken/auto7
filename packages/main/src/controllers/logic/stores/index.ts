export * from "./native-store"
export * from "./composed-store"
export * from "./LogicNative"
export * from "./LogicComposed"
