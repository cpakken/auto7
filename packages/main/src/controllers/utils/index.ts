export * from "./create-loader-store"
