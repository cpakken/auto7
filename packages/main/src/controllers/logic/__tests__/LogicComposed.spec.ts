import { getSnapshot } from "mobx-state-tree"
import { toJS } from "mobx"
import { ADDSTRING_COMPOUND } from "@main/test-utils/logic/composed"
import { createLogicComposedStoreSnapshot } from "@main/test-utils/logic/snapshot-factories"
import { ILogicComposedStore, LogicComposedStore } from "@main/controllers/logic"

describe("LogicComposed", () => {
  let store: ILogicComposedStore

  beforeEach(() => {
    const snapshot = createLogicComposedStoreSnapshot(ADDSTRING_COMPOUND)
    store = LogicComposedStore.create(snapshot)
  })

  test("store snapshot", () => {
    expect(getSnapshot(store)).toMatchSnapshot()
  })

  test("deleting io node will automatically delete connecting paths", () => {
    const { info, composition } = store.get("ADDSTRING_COMPOUND")

    info!.inputs.delete("in2")

    expect(toJS(composition!.blocks.get("A")!.inputs)).toMatchSnapshot()
    expect(toJS(composition!.inputs)).toMatchSnapshot()
    expect(getSnapshot(composition!).paths).toMatchSnapshot()
  })
})
