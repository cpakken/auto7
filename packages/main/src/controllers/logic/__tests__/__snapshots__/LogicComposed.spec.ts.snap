// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogicComposed deleting io node will automatically delete connecting paths 1`] = `
Map {
  "in1" => Array [
    Object {
      "block": undefined,
      "node": "in1",
      "path": Object {
        "_id": "p0",
        "from": Object {
          "node": "in1",
        },
        "to": Object {
          "block": "A",
          "node": "in1",
        },
      },
    },
  ],
  "in2" => Array [],
}
`;

exports[`LogicComposed deleting io node will automatically delete connecting paths 2`] = `
Map {
  "in1" => Array [
    Object {
      "block": Object {
        "_id": "A",
        "logicKey": "ADDSTRING",
        "xy": Array [
          0,
          0,
        ],
      },
      "node": "in1",
      "path": Object {
        "_id": "p0",
        "from": Object {
          "node": "in1",
        },
        "to": Object {
          "block": "A",
          "node": "in1",
        },
      },
    },
  ],
  "in2" => Array [],
  "in3" => Array [
    Object {
      "block": Object {
        "_id": "B",
        "logicKey": "ADDSTRING_NESTED",
        "xy": Array [
          0,
          0,
        ],
      },
      "node": "in2",
      "path": Object {
        "_id": "p3",
        "from": Object {
          "node": "in3",
        },
        "to": Object {
          "block": "B",
          "node": "in2",
        },
      },
    },
  ],
}
`;

exports[`LogicComposed deleting io node will automatically delete connecting paths 3`] = `
Array [
  Object {
    "_id": "p0",
    "from": Object {
      "node": "in1",
    },
    "to": Object {
      "block": "A",
      "node": "in1",
    },
  },
  Object {
    "_id": "p3",
    "from": Object {
      "node": "in3",
    },
    "to": Object {
      "block": "B",
      "node": "in2",
    },
  },
  Object {
    "_id": "p4",
    "from": Object {
      "block": "A",
      "node": "out1",
    },
    "to": Object {
      "block": "C",
      "node": "in1",
    },
  },
  Object {
    "_id": "p5",
    "from": Object {
      "block": "B",
      "node": "out1",
    },
    "to": Object {
      "block": "C",
      "node": "in2",
    },
  },
  Object {
    "_id": "p6",
    "from": Object {
      "block": "C",
      "node": "out",
    },
    "to": Object {
      "node": "out1",
    },
  },
]
`;

exports[`LogicComposed store snapshot 1`] = `
Object {
  "composition": Array [
    Object {
      "_id": "ADDSTRING_COMPOUND",
      "blocks": Array [
        Object {
          "_id": "A",
          "logicKey": "ADDSTRING",
          "xy": Array [
            0,
            0,
          ],
        },
        Object {
          "_id": "B",
          "logicKey": "ADDSTRING_NESTED",
          "xy": Array [
            0,
            0,
          ],
        },
        Object {
          "_id": "C",
          "logicKey": "_std_ADDSTRING",
          "xy": Array [
            0,
            0,
          ],
        },
      ],
      "paths": Array [
        Object {
          "_id": "p0",
          "from": Object {
            "node": "in1",
          },
          "to": Object {
            "block": "A",
            "node": "in1",
          },
        },
        Object {
          "_id": "p1",
          "from": Object {
            "node": "in2",
          },
          "to": Object {
            "block": "A",
            "node": "in2",
          },
        },
        Object {
          "_id": "p2",
          "from": Object {
            "node": "in2",
          },
          "to": Object {
            "block": "B",
            "node": "in1",
          },
        },
        Object {
          "_id": "p3",
          "from": Object {
            "node": "in3",
          },
          "to": Object {
            "block": "B",
            "node": "in2",
          },
        },
        Object {
          "_id": "p4",
          "from": Object {
            "block": "A",
            "node": "out1",
          },
          "to": Object {
            "block": "C",
            "node": "in1",
          },
        },
        Object {
          "_id": "p5",
          "from": Object {
            "block": "B",
            "node": "out1",
          },
          "to": Object {
            "block": "C",
            "node": "in2",
          },
        },
        Object {
          "_id": "p6",
          "from": Object {
            "block": "C",
            "node": "out",
          },
          "to": Object {
            "node": "out1",
          },
        },
      ],
    },
  ],
  "info": Array [
    Object {
      "_id": "ADDSTRING_COMPOUND",
      "inputs": Array [
        Object {
          "_id": "in1",
          "label": "in1",
          "typeKey": "_std_string",
        },
        Object {
          "_id": "in2",
          "label": "in2",
          "typeKey": "_std_string",
        },
        Object {
          "_id": "in3",
          "label": "in3",
          "typeKey": "_std_string",
        },
      ],
      "keysCache": undefined,
      "label": "ADDSTRING COMPOUND",
      "outputs": Array [
        Object {
          "_id": "out1",
          "label": "out1",
          "typeKey": "_std_string",
        },
      ],
    },
  ],
}
`;
